src/parser/parser.py

```diff
@@ -28,51 +39,72 @@ def _load_vdata(self):
         for file_name in os.listdir(self.DATA_DIR + scripts_path):
             if file_name.endswith('.json'):
                 # path/to/scripts/abilities.json -> abilities
-                key = file_name.split('.')[0].split('/')[-1]  
+                key = file_name.split('.')[0].split('/')[-1]
                 self.data['scripts'][key] = json_utils.read(
                     os.path.join(self.DATA_DIR, scripts_path, file_name)
                 )
 
     def _load_localizations(self):
-        names = json_utils.read(
-            self.DATA_DIR + 'localizations/gc/citadel_gc_' + self.language + '.json'
-        )
-        descriptions = json_utils.read(
-            self.DATA_DIR + 'localizations/mods/citadel_mods_' + self.language + '.json'
-        )
-        heroes = json_utils.read(
-            self.DATA_DIR + 'localizations/heroes/citadel_heroes_' + self.language + '.json'
-        )
-
-        self.localizations = {'names': names, 'descriptions': descriptions, 'heroes': heroes}
+        # Load all localizations data to memory
+
+        # Level 1: language
+        # Level 2: localization_group (i.e. attributes, gc, heroes, main, mods)
+        # Level 3: localization key
+        # Level 4: localized text
+        self.localizations = {}
+        for language in self.languages:
+            self.localizations[language] = {}
+            for localization_group in self.localization_groups:
+                localization_group_data = json_utils.read(
+                    self.DATA_DIR
+                    + 'localizations/'
+                    + localization_group
+                    + '/citadel_'
+                    + localization_group
+                    + '_'
+                    + language
+                    + '.json'
+                )
+                self.localizations[language][localization_group] = localization_group_data
 
     def run(self):
         print('Parsing...')
+        parsed_localizations = self._parse_localizations()
         parsed_abilities = self._parse_abilities()
         self._parse_heroes(parsed_abilities)
         self._parse_items()
         self._parse_changelogs()
         print('Done parsing')
 
+    def _parse_localizations(self):
+        print('Parsing Localizations...')
+        # TODO
+        # return localization.LocalizationsParser(
+        #     self.data['scripts']['localizations'],
+        #     self.localizations,
+        # ).run()
+
     def _parse_heroes(self, parsed_abilities):
         print('Parsing Heroes...')
         heroes.HeroParser(
             self.data['scripts']['heroes'],
             self.data['scripts']['abilities'],
             parsed_abilities,
-            self.localizations,
+            self.localizations[self.default_language],
```

Comment:
Why not a `language` attribute as you had before? That can  just default to `english`

Reply:
reverting